lib/webpacker/compiler.rb

```diff
@@ -53,11 +53,13 @@ def last_compilation_digest
 
     def watched_files_digest
       warn "Webpacker::Compiler.watched_paths has been deprecated. Set additional_paths in webpacker.yml instead." unless watched_paths.empty?
-      Dir.chdir File.expand_path(config.root_path) do
-        files = Dir[*default_watched_paths, *watched_paths].reject { |f| File.directory?(f) }
-        file_ids = files.sort.map { |f| "#{File.basename(f)}/#{Digest::SHA1.file(f).hexdigest}" }
-        Digest::SHA1.hexdigest(file_ids.join("/"))
+      root_path = File.expand_path(config.root_path)
+      expanded_paths = [*default_watched_paths, *watched_paths].map do |path|
+        File.join(root_path, path)
```

Comment:
There is a bug here if path and root_path are full paths. 

I fixed this here: 

https://github.com/rails/webpacker/pull/3233/files#diff-e9bc3ece054a6a66b7673121bfb015e820b756e2a0652de74befd08e2cf19386R66-R68